_gsocblogs/2022/blog_Rucio_NimishBongale.md

```diff
@@ -1,127 +1,279 @@
 
-## Personal Experience
+The DID Metadata page displays the different attributes associated to a DID, which a user can request to be shown by clicking an info button.
 
-It's been a great while so far contributing to the Rucio webui project under GSoC 2022. In the past couple of months, I've been able to understand Rucio, it's associated auth mechanisms, and complex workflows which are essential in developing a user-friendly UI. Learning from the deficits of the old webui has been key in bettering my understanding on an a comprehensive level. This has led to the development of a new up-to-the-standard UI library that is custom to Rucio while standing upon ui libraries like Bulma and Primer. Here is a gist of things I was able to do so far: <br />
+<img src="https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/DIDMetadata.png" alt="DID Metadata" width="100%">
 
-- [Feature 52 alert notifications](https://github.com/rucio/webui/pull/55)
+#### Rule Metadata Page
 
-#55 by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale) was merged 20 days ago• [Approved](https://github.com/rucio/webui/pull/55#partial-pull-merging)
+The Rule Metadata page displays the different attributes associated to a Rule, which a user can request to be shown by clicking on the scopename in blue.
 
-- [Feature 53 initial auth workflow](https://github.com/rucio/webui/pull/54)
+<img src="https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/RuleMetadata.png" alt="Rule Metadata" width="100%">
 
-#54 opened 25 days ago by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+is%3Aopen+author%3Animishbongale)
+#### List Rules Page
 
-- [Feature 47 login unit test cases](https://github.com/rucio/webui/pull/51)
+As the name suggests, this page enlists all the rules associated to a given account, and provides options to take actions on them. This is largely enabled by the Table component and different filters provided at the top. Also present on the view are buttons to delete and create new rules, which redirects us to the Rule Definition page.
 
-#51 opened 25 days ago by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+is%3Aopen+author%3Animishbongale)
+<img src="https://raw.githubusercontent.com/nimishbongale/webui/feature-add-snapshots/public/ListRules.png" alt="List Rules" width="100%">
 
-- [Create REST API wrapper utility](https://github.com/rucio/webui/pull/50)
+#### Rule Definition / Create Rules Page
 
-#50 by [nimishbongale](https://github.com/rucio/webui/issues?q=is%3Apr+author%3Animishbongale) was merged 26 days ago• [Approved](https://github.com/rucio/webui/pull/50#partial-pull-merging)
+Central to the functionality of the Rucio WebUI is the ability to create rules with just a few clicks.Once a rule is created, the rule automatically reflects in the List Rules page, owing to the implementation of the global store using the React Context API.
```

Comment:
We should also credit Valentina for laying the foundations of the error and rule workflow pages and the corresponding storybook components ;). You must mention here the work you have done on top to refactor it to the new architecture and clean up the UI to make it fit more with the rest of the components